### DISCUSSION TITLE
The unique rule from VineJS should be simpler?

### DISCUSSION THREAD

**Discussion by brunolipe-a** [adonisjs/core#4356] [Ideas]
I think that always use a query to check if the value is unique in database is to complex for something that we use very often.

Maybe we can have an uniqueRaw that uses the current code, and other that receives only table as required, something like that:

```ts
/**
 * Options accepted by the unique rule
 */
type Options = {
  table: string
  column?: string
  filter?: (db: DatabaseQueryBuilderContract, value: string, field: FieldContext) => Promise<void> 
}
```

**Comment by thetutlage** [comment]
Yes, it should be. Not sure, if you mentioned it also on Discord, or someone else did. But, I will explore some APIs to make it easier to work with

**Comment by thetutlage** [comment]
Moved here. https://github.com/adonisjs/auth/issues/233
